packages/upload-client/src/blob/add.js

```diff
@@ -165,19 +161,20 @@ export function createConcludeInvocation(id, serviceDid, receipt) {
  * has the capability to perform the action.
  *
  * The issuer needs the `blob/add` delegated capability.
+ * @param {import('multiformats').MultihashDigest} digest
  * @param {Blob|Uint8Array} data Blob data.
- * @param {import('./types.js').RequestOptions} [options]
- * @returns {Promise<import('./types.js').BlobAddOk>}
+ * @param {import('../types.js').RequestOptions} [options]
+ * @returns {Promise<import('../types.js').BlobAddOk>}
  */
 export async function add(
   { issuer, with: resource, proofs, audience },
+  digest,
```

Comment:
I think this needs to be updated: https://github.com/storacha-network/w3up/tree/main/packages/w3up-client#capabilityblobadd

Reply:
I did not change this... w3up client takes hash of blob.